lib/Twig/Filter.php

```diff
@@ -34,6 +34,10 @@ class Twig_Filter
      */
     public function __construct(string $name, $callable = null, array $options = array())
     {
+        if (__CLASS__ !== get_class($this)) {
+            @trigger_error('Overriding '.__CLASS__.' is deprecated since version 2.4.0 and the class will be final in 3.0.', E_USER_DEPRECATED);
```

Comment:
I think time has come. As we will soon have namespace support, I would want to release 3.0 soon after.